haskell/test/Lorentz/Contracts/Test/Management.hs

```diff
@@ -0,0 +1,504 @@
+-- SPDX-FileCopyrightText: 2020 tqtezos
+-- SPDX-License-Identifier: MIT
+
+-- | Tests for management entrypoints of stablecoin smart-contract
+
+module Lorentz.Contracts.Test.Management
+  ( managementSpec
+  ) where
+
+import Test.Hspec (Spec, describe, it)
+
+import Lorentz (mkView)
+import Lorentz.Contracts.Spec.FA2Interface as FA2
+import Lorentz.Contracts.Stablecoin
+import Lorentz.Contracts.Test.Common
+import Lorentz.Test
+import Util.Named
+
+managementSpec
+  :: forall param. ParameterC param
+  => OriginationFn param
+  -> Spec
+managementSpec originate = do
+  describe "Contract pausing" $ do
+    it "pauses contract as expected" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        lCallEP stablecoinContract (Call @"Pause") ()
+        validate . Right $ expectAnySuccess
+
+    it "cannot pause if sender does not have corresponding permissions" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        withSender wallet2 $ lCallEP stablecoinContract (Call @"Pause") ()
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "pause cannot be called multiple times in a row" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        lCallEP stablecoinContract (Call @"Pause") ()
+        lCallEP stablecoinContract (Call @"Pause") ()
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "unpauses contract as expected" $ integrationalTestExpectation $ do
+      let originationParams = defaultOriginationParams { opPaused = True }
+      withOriginated originate originationParams $ \stablecoinContract -> do
+        lCallEP stablecoinContract (Call @"Unpause") ()
+        validate . Right $ expectAnySuccess
```

Comment:
Shouldn't we also check the pause status was changed somehow?

Reply:
It seems that we have another test that check the above. So doesn't that make this test redundant?

Reply:
Done

Reply:
I cannot seem to find the corresponding change. What was it?